packages/core/src/execution-engine/workflow-execute.ts

```diff
@@ -354,6 +353,18 @@ export class WorkflowExecute {
 			`Could not find a node with the name ${destinationNodeName} in the workflow.`,
 		);
 
+		const dirtyNodes: Set<INode> = new Set();
+		for (const name of dirtyNodeNames) {
+			const node = workflow.getNode(name);
+			if (!node) {
+				console.warn(
+					`Could not find a node with the name ${name} in the workflow. This was passed in as a dirty node name.`,
+				);
+				continue;
+			}
+			dirtyNodes.add(node);
+		}
```

Comment:
Should we move this to the `Workflow` class instead? I'd prefer if all of the workflow state and any related utility methods were on that class.

Reply:
You mean something like `workflow.getNodes(names: sting[])`?

Reply:
If yes, should it fail with an assertion error if a node cannot be found, or should it log to the console, or do nothing?

Reply:
> You mean something like workflow.getNodes(names: sting[])?

Yes

> If yes, should it fail with an assertion error if a node cannot be found, or should it log to the console, or do nothing?

I think for now we should just log it to the console, since we already have a ghost nodes problem in the new canvas. After we fix those issues, we can switch to throwing an error.